src/Core/Form/IdentifiableObject/DataProvider/CombinationFormDataProvider.php

```diff
@@ -0,0 +1,100 @@
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataProvider;
+
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Query\GetCombinationForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\QueryResult\CombinationForEditing;
+use PrestaShop\PrestaShop\Core\Util\DateTime\DateTime;
+
+/**
+ * Provides the data that is used to prefill the Combination form
+ */
+class CombinationFormDataProvider implements FormDataProviderInterface
+{
+    /**
+     * @var CommandBusInterface
+     */
+    private $queryBus;
+
+    /**
+     * @param CommandBusInterface $queryBus
+     */
+    public function __construct(
+        CommandBusInterface $queryBus
+    ) {
+        $this->queryBus = $queryBus;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getData($id)
+    {
+        $combinationId = (int) $id;
+        /** @var CombinationForEditing $combinationForEditing */
+        $combinationForEditing = $this->queryBus->handle(new GetCombinationForEditing($combinationId));
+
+        return [
+            'id' => $combinationId,
+            'name' => $combinationForEditing->getName(),
+            'stock' => $this->extractStockData($combinationForEditing),
+        ];
+    }
+
+    /**
+     * @param CombinationForEditing $combinationForEditing
+     *
+     * @return array
+     */
+    private function extractStockData(CombinationForEditing $combinationForEditing): array
+    {
+        $stockInformation = $combinationForEditing->getStock();
+        $availableDate = $stockInformation->getAvailableDate();
+
+        return [
+            'quantity' => $stockInformation->getQuantity(),
+            'minimal_quantity' => $stockInformation->getMinimalQuantity(),
+            'stock_location' => $stockInformation->getLocation(),
+            'low_stock_threshold' => $stockInformation->getLowStockThreshold() ?: null,
+            'low_stock_alert' => $stockInformation->isLowStockAlertEnabled(),
+            'available_date' => $availableDate ? $availableDate->format(DateTime::DEFAULT_DATE_FORMAT) : '',
+        ];
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getDefaultData()
```

Comment:
```suggestion
    public function getDefaultData(): array
```

Reply:
No it inherit the doc which has `mixed` return type (no strong typing either) Please check the associated interfaces before making systematic request for missing types :)

Reply:
If it's not defined in interfaces, you can override in your implementation for improving extensibility.